### PR TITLE
finished task-3

### PR DESCRIPTION
**Pull request by DavidCobb606**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        return request(app).get("/api/topics").expect(200)
+
+    })
+    test("Responds with an array of topic objects which containg the properties `slug' and `description'", () =>{
+    return request(app)
+      .get("/api/topics")
+      .then(({body}) => {
+        
+        expect(body.topics).toBeInstanceOf(Array);
+        expect(typeof body).toBe("object");
```
Comment (two-triangles) [inline __tests__/api.test.js:26]: This isn't neccessary. The body that comes back from superTest will always be an object

File: api.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const express = require("express");
```
Comment (two-triangles) [inline api.js:1]: Possibly nitpicky - you have called your file api.js but your actual server is called app. I would rename your file to reflect this.

File: api.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+const app = express();
+
+app.get("/api/topics", getTopics);
+
+app.all("/*", (req,res) => {
+    res.status(404).send({msg: "Route not found"})
+})
+
+app.use((err, req, res, next) => {
```
Comment (two-triangles) [inline api.js:12]: You have an error handler that currently isn't doing anything - so I would either make this catch/debug errors, with a 500 status code as default.

File: controllers.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const {fetchTopics} = require("./models.js")
+
+exports.getTopics = (req,res, next) =>{
+    fetchTopics()
+    .then((topics) => {
+
+        console.log(topics)
```
Comment (two-triangles) [inline controllers.js:7]: remove this console log before merging

File: models.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+exports.fetchTopics = () => {
+    const command = `
+    SELECT *
+    FROM topics;
+    `;
+    return db.query(command)
+    .then(({rows: topics}) => {
+        return topics
+    })
+    .catch((err) => {
```
Comment (two-triangles) [inline models.js:12]: We don't need this catch block if we have one in the controller. Remember that your model returns a promise chain. The catch block in the controller will catch the error should this promise chain fail to fulfill

**Comment by two-triangles** [comment]
have a look over the comments and once you have amended, could you let me know and I will check before recommending merging

**Review by corkesw — COMMENTED** [review]
A couple of things to tidy up here and then you are good to merge

**Pull request merged**